src/main/java/com/example/todowithcouchbase/task/service/impl/TaskServiceImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.example.todowithcouchbase.task.service.impl;
+
+import com.example.todowithcouchbase.task.exception.TaskWithThisNameAlreadyExistException;
+import com.example.todowithcouchbase.task.model.dto.request.SaveTaskRequest;
+import com.example.todowithcouchbase.task.model.dto.response.SaveTaskResponse;
+import com.example.todowithcouchbase.task.model.entity.TaskEntity;
+import com.example.todowithcouchbase.task.model.mapper.SaveTaskRequestToTaskEntityMapper;
+import com.example.todowithcouchbase.task.model.mapper.TaskEntityToTaskResponseMapper;
+import com.example.todowithcouchbase.task.repository.TaskRepository;
+import com.example.todowithcouchbase.task.service.TaskService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class TaskServiceImpl implements TaskService {
+
+    private final TaskRepository taskRepository;
+    private final SaveTaskRequestToTaskEntityMapper saveTaskRequestToTaskEntityMapper =
+            SaveTaskRequestToTaskEntityMapper.initialize();
+
+    private final TaskEntityToTaskResponseMapper taskEntityToTaskResponseMapper=
```

Comment:
- Use TaskEntityToTaskMapper  here.

Reply:
The reason why I do not use this mapper is, as far as I know, it works to optimize the ddd model while generally using update, delete and insert operations. JVM opens a new object in the entityManager at the back and validates this transient object it opens and merges it after validation. In our case, we do not have any validation operations in the Task.

Reply:
In the service, It should return **task** domain object instead of **task response**.
Revise it.